hub/api/dataset.py

```diff
@@ -227,10 +235,94 @@ def like(
+
+        Returns:
+            Dataset: New dataset object with structured dataset.
+
+        Raises:
+            InvalidPathException: If the source directory does not exist.
+            SamePathException: If the source and destination path are same.
+        """
+        if not os.path.isdir(src):
+            raise InvalidPathException(src)
+
+        if os.path.isdir(src) and os.path.isdir(dest):
+            if os.path.samefile(src, dest):
+                raise SamePathException(src)
+
+        ds = hub.dataset(dest)
+
+        # TODO: support more than just image classification (and update docstring)
+        unstructured = ImageClassification(source=src)
+
+        # TODO: auto detect compression
+        unstructured.structure(
+            ds, image_tensor_args={"sample_compression": compression}  # type: ignore
+        )
+
+        return ds  # type: ignore
+
+    @staticmethod
+    @hub_reporter.record_call
+    def ingest_kaggle(
+        tag: str,
+        src: str,
+        dest: str,
+        src_creds: dict,
+        compression: str,
+        overwrite: bool = False,
+    ) -> Dataset:
+        """Download and ingest a kaggle dataset and store it as a structured dataset to destination
+
+        Note:
+            Currently only local source paths and image classification datasets are supported for automatic ingestion.
+
+        Args:
+            tag (str): Kaggle dataset tag. Example: `"coloradokb/dandelionimages"` points to https://www.kaggle.com/coloradokb/dandelionimages
+            src (str): Local path to where the unstructured dataset is stored.
+            dest (str): Destination path where the structured dataset will be stored. Can be:-
+                - a Hub cloud path of the form hub://username/datasetname. To write to Hub cloud datasets, ensure that you are logged in to Hub (use 'activeloop login' from command line)
+                - an s3 path of the form s3://bucketname/path/to/dataset. Credentials are required in either the environment or passed to the creds argument.
+                - a local file system path of the form ./path/to/dataset or ~/path/to/dataset or path/to/dataset.
+                - a memory path of the form mem://path/to/dataset which doesn't save the dataset but keeps it in memory instead. Should be used only for testing as it does not persist.
+            src_creds (dict): A dictionary containing credentials used to access the dataset at the path.
+            compression (str): Compression type of dataset.
+            overwrite (bool): WARNING: If set to True this overwrites the dataset if it already exists. This can NOT be undone! Defaults to False.
+
+        Returns:
+            Dataset: New dataset object with structured dataset.
+
+        Raises:
+            SamePathException: If the source and destination path are same.
+        """
```

Comment:
why aren't we checking the src path here similar to the first line in ingest?

Reply:
this method calls hub.ingest